src/main/java/com/example/usermanagement/config/SecurityConfig.java

```diff
@@ -1,40 +1,53 @@
 package com.example.usermanagement.config;
 
-import lombok.RequiredArgsConstructor;
+import org.keycloak.adapters.AdapterDeploymentContext;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.security.authentication.AuthenticationProvider;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.oauth2.server.resource.OAuth2ResourceServerConfigurer;
 import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.session.SessionRegistryImpl;
 import org.springframework.security.web.SecurityFilterChain;
-import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.security.web.authentication.session.RegisterSessionAuthenticationStrategy;
+import org.springframework.security.web.authentication.session.SessionAuthenticationStrategy;
 
 
 @Configuration
 @EnableWebSecurity
-@RequiredArgsConstructor
-public class SecurityConfig {
+public class SecurityConfig{
 
 
-    private final JwtAuthenticationFilter jwtAuthFilter;
-    private final AuthenticationProvider authenticationProvider;
 
+    @Bean
+    protected SessionAuthenticationStrategy sessionAuthenticationStrategy(){
```

Comment:
why do we need this bean?

Reply:
This bean in Keycloak configuration is used to define the strategy for handling session authentication. It determines how session-related authentication events are processed and managed within Keycloak.